spring-transaction/src/main/kotlin/org/jetbrains/exposed/spring/SpringTransactionManager.kt

```diff
@@ -137,6 +196,23 @@ class SpringTransactionManager(
         }
 
         trxObject.setCurrentToOuter()
+
+        if (trxObject.isNewConnectionHolder) {
+            TransactionSynchronizationManager.unbindResource(dataSource)
+            trxObject.getConnectionHolder().released()
+        }
+        trxObject.getConnectionHolder().clear()
+
+        // Reset connection.
+        val con: Connection = trxObject.getConnectionHolder().getConnection()
+        try {
+            if (trxObject.mustRestoreAutoCommit) {
+                con.autoCommit = true
+            }
+            DataSourceUtils.resetConnectionAfterTransaction(con, trxObject.previousIsolationLevel, trxObject.isReadOnly)
+        } catch (ex: Throwable) {
+            logger.debug("Could not reset JDBC Connection after transaction", ex)
```

Comment:
Some logging came with and ended up identical to the Spring JDBC ones. Obviously those should be cleaned up.